### ISSUE TITLE
member-access false positive for functions in ES6 object literals

### ISSUE THREAD

**Issue by nikklassen** [palantir/tslint#612]
When ```member-access``` is true defining a function using the ES6 object literal shorthand (found in section 4.5 of the Typescript [Language Specification](http://www.typescriptlang.org/Content/TypeScript%20Language%20Specification.pdf)) causes an error ("default access modifier on member/method is not allowed").

```ts
const o = {
    foo() {
        return 0;
    }
};
```

However, it is not valid to add a AccessModifier to object literals.

**Comment by nikklassen** [comment]
I can take a crack at fixing this.

**Comment by Strate** [comment]
:+1:
